### PR TITLE
model: change all the names to move towards abstract build graph

### PR DESCRIPTION
**Pull request by nicks**
Hello @maiamcc, @jazzdan,

Please review the following commits I made in branch nicks/targets:

74ae9cb7e9559e39ce87758da9cbd2761d46318a (2019-01-09 16:38:26 -0500)
model: change all the names to move towards abstract build graph

### CODE REVIEW

File: internal/model/target.go [hunk 1]
```diff
@@ -0,0 +1,29 @@
+package model
```
Comment (nicks) [inline internal/model/target.go]: the big change is this file. everything else is just renaming

**Comment by nicks** [comment]
going to merge this now to avoid merge conflicts but will follow-up on any other revisions on names!

File: internal/model/target.go [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+
+// An abstract build graph of targets and their dependencies.
+// Each target should have a unique ID.
+
+type TargetType string
+
+const (
+	TargetTypeK8s           TargetType = "k8s"            // Deployed k8s entities
+	TargetTypeImage         TargetType = "image"          // Image builds
+	TargetTypeDockerCompose TargetType = "docker-compose" // Docker-compose service build+deploy
```
Comment (maiamcc) [inline internal/model/target.go]: worth including in the note that we treat it as a deploy target right now but that that's sort of a hack

Comment (nicks) [reply internal/model/target.go]: done in the next pr

**Pull request merged**